js/core/utils/number_parser_generator.js

```diff
@@ -116,4 +117,63 @@ function generateNumberParser(format) {
     };
 }
 
+var formatRules = {
+    "#": {
+        regexp: "\\d*",
+        empty: ""
+    },
+
+    "0": {
+        regexp: "\\d",
+        empty: "0"
+    }
+};
+
+function getFormatString(format, value) {
+    value = value || 0;
+
+    var stringValue = value.toString(),
+        specialFormatChars = Object.keys(formatRules),
+        resultString = "";
+
+    for(var i = 0; i < format.length; i++) {
+        var formatChar = format.charAt(i);
+
+        if(specialFormatChars.indexOf(formatChar) === -1) {
+            resultString += formatChar;
+            stringValue = stringValue.replace(new RegExp("^" + formatChar), "");
+            continue;
+        }
+
+        var formatRule = formatRules[formatChar],
+            formatRegExp = new RegExp("^" + formatRule.regexp);
+
+        var matches = stringValue.match(formatRegExp);
+
+        resultString += matches ? matches[0] : formatRule.empty;
+        stringValue = stringValue.replace(formatRegExp, "");
+    }
+
+    return resultString;
+}
+
+function generateNumberFormatter(format) {
```

Comment:
Move this method to another file or rename current file. The number_parser_generator file name is not good for formatter.